### PR TITLE
Reorganizing the functions

### PR DESCRIPTION
**Pull request by matthiasto**
Reorganizing the different functions
Adding feature flag
Looking at the SPH

### CODE REVIEW

File: avaframe/com1DFAPy/DFAtools.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        Nz[0, 0] = 1
+        Nx[n-1, 0] = -(z[n-1, 1] - z[n-1, 0]) / csz
+        Ny[n-1, 0] = (z[n-2, 0] - z[n-1, 0]) / csz
+        Nz[n-1, 0] = 1
+        Nx[0, m-1] = (z[0, m-2] - z[0, m-1]) / csz
+        Ny[0, m-1] = -(z[1, m-1] - z[0, m-1]) / csz
+        Nz[0, m-1] = 1
+        Nx[n-1, m-1] = (z[n-1, m-2] - z[n-1, m-1]) / csz
+        Ny[n-1, m-1] = (z[n-2, m-1] - z[n-1, m-1]) / csz
+        Nz[n-1, m-1] = 1
```
Comment (fso42) [inline avaframe/com1DFAPy/DFAtools.py:83]: Please make an issue out of this: "Try to optimize for readbility" but don't adress this in this PR

File: avaframe/com1DFAPy/SPHfunctions.py [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+    L = np.delete(L, indJ)
+    dx = particles['x'][L] - x
+    dy = particles['y'][L] - y
+    dz = (particles['z'][L] - z)
+    r = DFAtls.norm(dx, dy, dz)
+    # impose a minimum distance between particles
+    dx = np.where(r < 0.0001 * rKernel, 0.0001 * rKernel * dx, dx)
+    dy = np.where(r < 0.0001 * rKernel, 0.0001 * rKernel * dy, dy)
+    dz = np.where(r < 0.0001 * rKernel, 0.0001 * rKernel * dz, dz)
+    r = np.where(r < 0.0001 * rKernel, 0.0001 * rKernel, r)
```
Comment (fso42) [inline avaframe/com1DFAPy/SPHfunctions.py:212]: Should we put this distance somewhere as parameter, or is 0.0001 a commonly accepted value?

File: avaframe/com1DFAPy/SPHfunctions.py [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+    W11 = facKernel * hr11 * hr11 * hr11
+    f11 = f * m * W11 / rho
+    np.add.at(F, ic11, f11)
+
+    F = np.reshape(F, (nrow, ncol))
+
+    return F
+
+
+def computeFD(cfg, particles, dem):
```
Comment (fso42) [inline avaframe/com1DFAPy/SPHfunctions.py:328]: Maybe a more descriptive name?

File: avaframe/com1DFAPy/com1DFA.py [hunk 4]
```diff
@@ -1,195 +1,1158 @@
...
+        else:
+            particles, fields, Tcpu = computeTimeStep(cfg, particles, fields, dt, dem, Ment, Cres, Tcpu)
+        # Save desired parameters and export to Lists for saving interval
+        U = np.append(U, DFAtls.norm(particles['ux'][0], particles['uy'][0], particles['uz'][0]))
+        Z = np.append(Z, particles['z'][0])
+        S = np.append(S, particles['s'][0])
+        iterate = particles['iterate']
+        if t >= nSave * dtSave:
+            log.info('Saving results for time step t = %f s', t)
+            log.info('MTot = %f kg, %s particles' % (particles['mTot'], particles['Npart']))
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:270]: could this be debug level?

File: avaframe/com1DFAPy/com1DFA.py [hunk 5]
```diff
@@ -1,195 +1,1158 @@
...
+        gravAccTangZ = -gravAcc - gravAccNorm * nzAvg
+        # adding gravity force contribution
+        forceX[j] = forceX[j] + gravAccTangX * mass
+        forceY[j] = forceY[j] + gravAccTangY * mass
+        forceZ[j] = forceZ[j] + gravAccTangZ * mass
+
+        # Calculating bottom shear and normal stress
+        if(effAccNorm > 0.0):
+            # if fluid detatched
+            log.info('fluid detatched for particle %s', j)
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:581]: debug level?

File: avaframe/com1DFAPy/com1DFA.py [hunk 6]
```diff
@@ -1,195 +1,1158 @@
...
+    # kinEneNew = np.sum(0.5 * massNew * norm2(particles['ux'], particles['uy'], particles['uz']))
+    # potEneNew = np.sum(gravAcc * massNew * particles['z'])
+    # totEneNew = kinEneNew + potEneNew
+    # log.info('total energy variation: %f' % ((totEneNew - totEne) / totEneNew))
+    # log.info('kinetic energy variation: %f' % ((kinEneNew - kinEne) / kinEneNew))
+    # if (abs(totEneNew - totEne) / totEneNew < 0.01) and (abs(kinEneNew - kinEne) / kinEneNew < 0.01):
+    #     log.info('Reached stopping creteria : total energy varied fom less than 1 %')
+    #     particles['stoppCriteria'] = True
+    # particles['kineticEne'] = kinEneNew
+    # particles['potentialEne'] = potEneNew
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:888]: Are these still needed?

File: avaframe/com1DFAPy/com1DFA.py [hunk 7]
```diff
@@ -1,195 +1,1158 @@
...
+    # VZSPH = pointsToRasterSPH(particles, rho, Z, uz, VZSPH, csz=csz)
+    # VSPH = norm(VXSPH, VYSPH, VZSPH)
+    # FDSPH = hSPH
+    # # FDSPH = MassSPH / (A * rho)
+    # PSPH = VSPH * VSPH * rho
+    # # PV = np.where(VSPH > PV, VSPH, PV)
+    # # PP = np.where(PSPH > PP, PSPH, PP)
+    # # PFD = np.where(FDSPH > PFD, FDSPH, PFD)
+
+    #########################################
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:933]: Still needed?

File: avaframe/com1DFAPy/com1DFA.py [hunk 8]
```diff
@@ -1,195 +1,1158 @@
...
+    # # startTime = time.time()
+    MassBilinear = geoTrans.pointsToRaster(x, y, m, MassBilinear, csz=csz, interp='bilinear')
+    MomBilinearX = geoTrans.pointsToRaster(x, y, m * ux, MomBilinearX, csz=csz, interp='bilinear')
+    MomBilinearY = geoTrans.pointsToRaster(x, y, m * uy, MomBilinearY, csz=csz, interp='bilinear')
+    MomBilinearZ = geoTrans.pointsToRaster(x, y, m * uz, MomBilinearZ, csz=csz, interp='bilinear')
+
+    # VXBilinear = np.where(MassBilinear > 0, MomBilinearX/MassBilinear, MomBilinearX)
+    # VYBilinear = np.where(MassBilinear > 0, MomBilinearY/MassBilinear, MomBilinearY)
+    # VZBilinear = np.where(MassBilinear > 0, MomBilinearZ/MassBilinear, MomBilinearZ)
+    # VBilinear = norm(VXBilinear, VYBilinear, VZBilinear)
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:949]: If condition needed to switch code?

File: avaframe/com1DFAPy/com1DFA.py [hunk 9]
```diff
@@ -1,195 +1,1158 @@
...
+    cmap.set_under(color='w')
+    ref0, im = NonUnifIm(ax, xx, yy, data, 'x [m]', 'y [m]',
+                         extent=[x.min(), x.max(), y.min(), y.max()],
+                         cmap=cmap, norm=norm)
+    if plotPart:
+        ax.plot(x, y, 'ok', linestyle='None', markersize=1)
+    Cp1 = ax.contour(X, Y, Z, levels=10, colors='k')
+    addColorBar(im, ax, ticks, unit)
+    plt.pause(0.1)
+    # plt.close(fig)
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:1071]: This is needed, right?

File: avaframe/com1DFAPy/com1DFA.py [hunk 10]
```diff
@@ -1,195 +1,1158 @@
...
+    B = cfg.getfloat('B')
+    R = cfg.getfloat('R')
+    Rs = rho * v * v / (p + 0.001)
+    div = h / R
+    div = np.where(div < 1.0, 1, div)
+    # if(div < 1.0):
+    #     div = 1.0
+    div = np.log(div) / kappa + B
+    tau = p * mu * (1.0 + Rs0 / (Rs0 + Rs)) + rho * v * v / (div * div)
+    return tau
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:1158]: Move this function to a file frictionLaws.py? -> not necessarily in this PR

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com1DFAPy/DFAtools.py (hunk 1 above)
Comment (matthiasto) [reply avaframe/com1DFAPy/DFAtools.py:83]: I created one regarding the normal calculation

File: avaframe/com1DFAPy/SPHfunctions.py (hunk 2 above)
Comment (matthiasto) [reply avaframe/com1DFAPy/SPHfunctions.py:212]: also created one yesterday

File: avaframe/com1DFAPy/com1DFA.py (hunk 9 above)
Comment (matthiasto) [reply avaframe/com1DFAPy/com1DFA.py:1077]: the close? or the pause?

File: avaframe/com1DFAPy/SPHfunctions.py (hunk 2 above)
Comment (matthiasto) [reply avaframe/com1DFAPy/SPHfunctions.py:216]: # 204

File: avaframe/com1DFAPy/com1DFA.py (hunk 9 above)
Comment (fso42) [reply avaframe/com1DFAPy/com1DFA.py:1077]: The close

File: avaframe/com1DFAPy/DFAtools.py (hunk 1 above)
Comment (matthiasto) [reply avaframe/com1DFAPy/DFAtools.py:83]: issue #199

File: avaframe/com1DFAPy/com1DFA.py (hunk 4 above)
Comment (matthiasto) [reply avaframe/com1DFAPy/com1DFA.py:270]: maybe I leave this one as info for now, I switched the 2 previous prints as debug

File: avaframe/com1DFAPy/com1DFA.py (hunk 9 above)
Comment (matthiasto) [reply avaframe/com1DFAPy/com1DFA.py:1048]: I just update the figure so I dont want to close it

**Pull request merged**